tests/test_benchmarks.py

```diff
@@ -0,0 +1,433 @@
+        _validate_benchmark_file(benchmark_file, benchmark_module)
+
+
+class TestLoadBenchmark:
+
+    def test_load_benchmark(self, valid_single_benchmark):
+        benchmark = _benchmarks.Benchmark.from_dict(**valid_single_benchmark)
+        assert benchmark.serializable() == valid_single_benchmark
+
+    def test_load_benchmark_set_dialect(self, valid_single_benchmark):
+        valid_single_benchmark["schema"][
+            "$schema"
+        ] = Dialect.latest().serializable()
+        benchmark = _benchmarks.Benchmark.from_dict(
+            **valid_single_benchmark,
+        ).maybe_set_dialect_from_schema()
+        assert benchmark.dialect == Dialect.latest()
+
+    def test_load_benchmark_with_diff_tests(self, valid_single_benchmark):
+        benchmark = _benchmarks.Benchmark.from_dict(
+            **valid_single_benchmark,
+        ).benchmark_with_diff_tests(
+            tests=valid_single_benchmark["tests"] * 10,
+        )
+        assert benchmark_validated(benchmark.serializable())
+
+    def test_load_single_benchmark_group_from_dict(
+        self,
+        valid_single_benchmark,
+    ):
+        benchmark = valid_single_benchmark
+        benchmark_group = BenchmarkGroup.from_dict(benchmark)
+
+        assert benchmark_validated(benchmark_group.serializable())
+
+    def test_load_benchmark_group_from_dict(self, valid_grouped_benchmark):
+        benchmark = valid_grouped_benchmark
+        benchmark_group = BenchmarkGroup.from_dict(benchmark)
+
+        serializable = benchmark_group.serializable()
+
+        assert benchmark == serializable
+        assert benchmark_validated(serializable)
+
+    def test_load_single_benchmark_group_from_json(
+        self,
+        tmp_path,
+        valid_single_benchmark,
+    ):
+        tmp_path = tmp_path / "test_file.json"
+        tmp_path.write_text(json.dumps(valid_single_benchmark))
+        benchmark_group = BenchmarkGroup.from_file(tmp_path)
+        assert benchmark_validated(benchmark_group.serializable())
+
+    def test_load_benchmark_group_from_json(
+        self,
+        tmp_path,
+        valid_grouped_benchmark,
+    ):
+        tmp_path = tmp_path / "test_file.json"
+        tmp_path.write_text(json.dumps(valid_grouped_benchmark))
+        benchmark_group = BenchmarkGroup.from_file(tmp_path)
+
+        serializable = benchmark_group.serializable()
+
+        assert serializable == valid_grouped_benchmark
+        assert benchmark_validated(serializable)
+
+    def test_load_benchmark_groups_from_folder(self):
+        benchmark_groups = BenchmarkGroup.from_folder(
+            default_benchmarks_dir,
+            module="bowtie.benchmarks",
+        )
+        for benchmark_group in benchmark_groups:
+            assert benchmark_validated(benchmark_group.serializable())
+
+
+class TestBenchmarker:
+
+    def test_default_benchmarker(self, benchmarker_run_args):
+        _benchmarks.Benchmarker.from_default_benchmarks(**benchmarker_run_args)
```

Comment:
Is this meant to test it doesn't blow up with those arguments, or what?

Reply:
Yeah just testing if the benchmarker loads up fine. So that in case something blows up we know where its blowing up.